src/wp-includes/class-wp-navigation-fallback.php

```diff
@@ -26,9 +26,18 @@ class WP_Navigation_Fallback {
 	 */
 	public static function get_fallback() {
 
+		/**
+		 * Filters whether or not a fallback should be created.
+		 *
+		 * @since 6.3.0
+		 *
+		 * @param bool Whether or not to create a fallback.
+		 */
+		$should_create_fallback = apply_filters( 'wp_navigation_should_create_fallback', true );
+
 		$fallback = static::get_most_recently_published_navigation();
 
-		if ( $fallback ) {
+		if ( $fallback || ! $should_create_fallback ) {
```

Comment:
Note: we're not opting out of fetching a fallback, it's opting out of creating one either from:

- converting a classic menu
- creating a new block based menu